### ISSUE TITLE
"content" attribute not honoured on all tags

### ISSUE THREAD

**Issue by iamjochem** [Janpot/microdata-node#6]
hi there,

the module used to parse [microdata](https://github.com/Janpot/microdata-node) does not honour the microdata spec with regard to "**content**" tags attributes.

[the spec](https://www.w3.org/TR/microdata/#values)  states: 

> HTML only allows the content attribute on the meta element. This specification changes the content model to allow it on any element, as a global attribute.

[the relevant function](https://github.com/Janpot/microdata-node/blob/master/lib/parse.js#L111) in microdata-node only looks at the "**content**" attribute for "**meta**" tags.

It is understandable that only **meta** tags are considered given the underlying HTML spec, but I have noticed a lot of real-life HTML along the following lines (i.e. regardless of correctness this lib would be even more useful if it always grabbed a "**content**" attribute, if it is set):

```html
<span itemProp="someprop" content="somevalue">this is not the content</span>
<span itemProp="anotherprop" content="anothervalue" /><!-- self closing SPANs (etc) are invalid but found in the wild none the less! -->
```

**N.B.** I believe it is not enough to fallback to trying a "**content**" attribute if no inner tag text is found ... the "**content**" attribute should take precedence over inner tag text if it exists (in the HTML example above I believe the property value should be "_somevalue_" rather than "_this is not the content_")

> I noticed this issue whilst using a [dependent module, html-metadata](https://github.com/wikimedia/html-metadata) (reference to downstream issue: https://github.com/wikimedia/html-metadata/issues/68)

thanks for reading, interested to know what your opinion is.

**Comment by Janpot** [comment]
You are correct. At first sight it looks like [it got added](https://w3c.github.io/microdata/#changes) to newer versions of the spec.
I'll have to revise those changes when I have some time.
I'd also accept a pull request if you're interested.

**Comment by iamjochem** [comment]
hi, 

glad you agree, I think I can manage a PR but I'll need some more input from you:

my premise is in contradiction with code with regard to the current functionality of the `strict` parameter/option, how do you think it should work? personally I feel that `strict` option should be removed completely, with the lib behave in a non-strict manner in addition to implementing the strategy I suggested above.

obivously dropping the `strict` option and changing the behaviour is a breaking change - is that something you're okay with? (you're using semver so I guess it would imply moving to a 1.x

**Comment by Janpot** [comment]
"strict" means "parses strictly according to spec". So I don't really see why it should be removed. Its behavior needs to adapted to the latest spec changes.
In hindsight, I think I want to have a quick look myself first, it's been a while since I went through this code and I want to bring the repo to 2018 first.

**Comment by iamjochem** [comment]
okay cool - let me know if/when you come to a conclusion - I'm sure I can put together a PR based on your feedback.
